go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -171,33 +162,36 @@ func (bnc *BaseNetworkController) ensurePod4SecondaryNetwork(pod *kapi.Pod, addP
 
 // removePod tried to tear down a pod. It returns nil on success and error on failure;
 // failure indicates the pod tear down should be retried later.
-func (bnc *BaseNetworkController) removePod4SecondaryNetwork(pod *kapi.Pod, portInfo *lpInfo) error {
-	var nadName, logicalPortName string
+func (bnc *BaseNetworkController) removePod4SecondaryNetwork(pod *kapi.Pod, portInfoMap map[string]*lpInfo) error {
+	var nadName string
+	var portInfo *lpInfo
 
-	if !util.PodWantsNetwork(pod) {
-		return nil
-	}
 	podDesc := pod.Namespace + "/" + pod.Name
 	klog.Infof("Deleting pod: %s for network %s", podDesc, bnc.GetNetworkName())
 
-	if !util.PodScheduled(pod) {
-		return nil
-	}
+	if portInfoMap != nil {
+		if len(portInfoMap) == 0 {
+			return nil
+		}
+		for nadName, portInfo = range portInfoMap {
```

Comment:
I am bit confused that we added support in the port_cache to hold port info per NAD but that we only use the first one here because there's only actually going to be one. Is this in preparation for something coming next? Can you elaborate?

Reply:
yes, as in downstream (and very probably in later upstream), we are going to support multiple nads per secondary network

Reply:
OK, I was wondering if it would be an actual need for it in this PR. I would probably not have done it in this PR though if there is no actually need, we are just making a PR that is already big just bigger.